### ISSUE TITLE
DB access patterns: Make `pread` the default (instead of `mmap`)

### ISSUE THREAD

**Issue by etiennedi** [weaviate/weaviate#4464]
A previous release introduced the option to use `pread` instead of `mmap`. We have since already run `pread` by default on the Weaviate Cloud Service and are happy with it. Initial benchmarks show that there is no disadvantage to using `pread`. In many situations it outperforms `mmap`, especially under highly concurrent load.

## TODOs
- [ ] The current benchmarks are somewhat anecdotal. We should formalize them be able to make a data-driven decision.
- [ ] If we are happy with the results, the next release (e.g. `v1.25` could default to using `pread`)

**Comment by daveatweaviate** [comment]
What is the advantage of using one or the other? The docs say [this](https://weaviate.io/developers/weaviate/configuration/persistence#virtual-memory-access-method): 

`mmap uses a memory-mapped file, which means that the file is mapped into the virtual memory of the process. pread is a function that reads data from a file descriptor at a given offset.

In general, mmap may be a preferred option with memory management benefits. However, if you experience stalling situations under heavy memory load, we suggest trying pread instead.`

Whichever one we go with, we might want to expand on the differences.

**Comment by etiennedi** [comment]
Hi @daveatweaviate, apologies, I originally missed your comment.

The tl;dr is:
- There is some good research (done by our friends at CMU) indicating that `mmap` is a bad idea for databases in genereal. This [video](https://www.youtube.com/watch?v=1BRGU_AS25c) is a great summary
- In addition to the general mmap topics, Go is quite bad when it comes to `mmap`. In essence, if many goroutines are IO-blocked from mmap, the runtime doesn't find out about it and thinks the routines are doing actual work. Therefore, it does not pause their execution. This means that other routines may be scheduled. In practice this typically leads to timeouts, stalling behavior, etc. 
- All WCS clusters already run with `pread`, so it's only community users and self-hosters that are currently still defaulting to `mmap`.

Side-note: The `pread` implementation was only ~90% complete until now. There were some cases where we still fell back to `mmap`. The goal is to get that to 100% in v1.25. This could be an argument to delay the switch to v1.26, so we have one more release to gain experience with the updated implementation.
